pkg/analyzer/detection/detection.go

```diff
@@ -27,8 +27,12 @@ func DetectComponents(name string, files analyzer.Files, parent *database.Layer,
 	var featureVersions []database.FeatureVersion
 	var rhelfeatures *database.RHELv2Components
 
-	if namespace != nil && wellknownnamespaces.IsRHELNamespace(namespace.Name) {
-		// This is a RHEL-based image that must be scanned in a certified manner.
+	// In the current state, RHCOS will always be handled as certified system,
+	// no matter what value uncertifiedRHEL is.
+	// This is in contrast to RHEL, where uncertifiedRHEL determines whether we scan for RHEL or CentOS.
+	// TODO(ROX-13906): Verify RHCOS logic
+	if namespace != nil && (wellknownnamespaces.IsRHELNamespace(namespace.Name) || wellknownnamespaces.IsRHCOSNamespace(namespace.Name)) {
```

Comment:
Maybe for another time, but perhaps we can make something like `UseCertifiedWorkflow` which checks if the namespace is `rhel` or `rhcos`. Minimizes all the `this || that`

Reply:
That is a good idea!
I have added a follow up ticket (ROX-14028) to track the refactoring.